### DISCUSSION TITLE
[Bug]: Instead of pimcore login trying to implementing with Azure loin flow Facing the ERR_TOO_MANY_REDIRECTS issue

### DISCUSSION THREAD

**Discussion by manasaR507** [pimcore/pimcore#12649] [Q&A / Support]
### Expected behavior

Hi Pimcore team,

We are currently implemented azure ad login instead of using with pimcore login flow. I had implemented with Azure login flow. 
When azure user logged in successfully. it should redirected to  admin index page. Admin index page load properly.

### Actual behavior

1. When user hits the /admin redirected to  azure login page. 
2. If a new user, We are registering automatically in users table and assigning 'default_Roles' or else for old user redirecting /admin. Below, $link URL pattern using for admin redirection.

        ` // login as user
            $token = Tool\Authentication::generateToken($userName, $userPassword);
            $link = $request->getScheme() . '://' . $request->getHttpHost() . '/admin/login/login?username=' . $userName . '&token=' . $token;

            return $this->redirect($link);`

3. Actually issue facing here - When we are login into the browser first time. After azure user logged in successfully. Facing the issue with first time admin page load. Throwing the 'ERR_TOO_MANY_REDIRECTS' error. Attached screenshot below. After reloading the page. Working fine. Successfully redirecting to admin index page
![Screenshot (191)](https://user-images.githubusercontent.com/64125570/178926046-a10c5740-a870-461d-88db-68260568011a.png)
![Screenshot (189)](https://user-images.githubusercontent.com/64125570/178927257-31507762-0187-4276-be4a-e68fa05749ab.png)
.

### Steps to reproduce

1.  In our custom bundle, Copied pimcore admin logincontroller.php under the CustomBundle/Admin/LoginController.php. Here /login action, Made few changes, added the micorosft auth ur redirection.
2. In Routing.yaml added admin login route & azure redirection url route. Observe below
```
//Pimcore admin login
pimcore_admin_login:
    path: /login
    controller:  App\CustomBundle\Controller\Admin\LoginController::loginAction
 // Azure Redirection url
admin_auth_oauth_check:
    path: /auth/azure
    controller: App\CustomBundle\Controller\AzureController::authCheckAction
```
3. When user hit the /admin redirecting to microsoft login page. After giving valid credentials. Sucessfuly getting azure user info and redirecting to pimcore admin index page. With below pattern.
 
```
// login as user
$token = Tool\Authentication::generateToken($userName, $userPassword);
$link = $request->getScheme() . '://' . $request->getHttpHost() . '/admin/login/login?username=' . $userName . '&token=' . $token;
return $this->redirect($link);
```

5. For the first time user logged in the browser. Getting the error like **{maindomain} redirected you too many times**.
6. If already user logged in the browser  working fine. successfully redirecting to admin index page. 

Can anyone have the idea. Please give me some guidance for fixing this issue.

**Comment by skynet-technologies** [comment]
It might be the cache problem. Have you tried clearing it?
Also, check if the login session is generated or not.

**Reply by manasaR507** [reply]
Yes, I had done cache clear and  session is generated. Both are fine and also cleared browser cache too. But, still facing the same issue.
